src/test/java/fi/aalto/cs/intellij/common/VersionTest.java

```diff
@@ -0,0 +1,134 @@
+package fi.aalto.cs.intellij.common;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+
+import org.junit.Test;
+
+public class VersionTest {
+
+  @Test
+  public void testCreateVersion() {
+    Version version = new Version(3,5,24);
+
+    assertEquals(3, version.major);
+    assertEquals(5, version.minor);
+    assertEquals(24, version.build);
+
+    assertEquals("3.5.24", version.toString());
+
+    Version sameVersion = new Version(3, 5, 24);
+    Version differentVersion = new Version(4, 6, 78);
+
+    assertEquals(version, sameVersion);
+    assertEquals(version.hashCode(), sameVersion.hashCode());
+    assertNotEquals(version, differentVersion);
+  }
+
+  @Test
+  public void testCreateVersionWithNegativeMajorVersion() {
+    try {
+      new Version(-1, 4, 5);
+    } catch (IllegalArgumentException ex) {
+      return;
+    }
+    fail();
+  }
+
+  @Test
```

Comment:
could this approach be applicable when working with exceptions? 
```java
@Test(expected = NullPointerException.class)
public void whenExceptionThrown_thenExpectationSatisfied() { 
...
```

Reply:
Yes if the properties of the exception (message, cause, etc.) don't have to be checked. So in this case that would be a good way to go. There is also ExpectedException rule in JUnit but to me, it seemed to add little value.

Junit 5 provides even more sophisticated solutions but I think testing exceptions this way is quite ok. The biggest issue is that `fail()` imay be easily forgotten.